### PR TITLE
[SLS-595] Update layer to use new tracing format

### PR DESCRIPTION
**Pull request by hghotra**
### What does this PR do?

Update layer to use new tracing format.

### Motivation

https://datadoghq.atlassian.net/browse/SLS-595

### Testing Guidelines

Tested manually.

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [x] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: src/handler.ts [hunk 1]
```diff
@@ -5,7 +5,11 @@ const { load } = require("/var/runtime/UserFunction") as any;
 
 if (getEnvValue("DD_TRACE_ENABLED", "true").toLowerCase() === "true") {
   // tslint:disable-next-line:no-var-requires
-  require("dd-trace").init();
+  require("dd-trace").init({
```
Comment (DarcyRaynerDD) [inline src/handler.ts:8]: This only applies for the redirected handler. It won't apply if they are using the wrapper manually.

Comment (hghotra) [reply src/handler.ts:8]: If they use the wrapper manually, then the forwarder seems to take care of it.

File: src/trace/listener.ts [hunk 2]
```diff
@@ -111,7 +114,9 @@ export class TraceListener {
       options.childOf = spanContext;
     }
     options.type = "serverless";
-    options.resource = this.handlerName;
+    options.service = "aws.lambda";
+    options.resource = this.context && this.context.functionName;
```
Comment (DylanLovesCoffee) [inline src/trace/listener.ts:118]: In the case that `this.context` is non-existent/falsy, would `resource` display the falsy value as a string?

Comment (hghotra) [reply src/trace/listener.ts:118]: What should the `resource` be if the `context` is `undefined`? Currently, `context` gets initialized here: https://github.com/DataDog/datadog-lambda-js/blob/b64a50e3316fff7df9d0d17ae56db06180690262/src/trace/listener.ts#L61
After the initialization, if `context` is `undefined`, then resource will get set to `undefined` - maybe we shouldn't set it at all?

Comment (DylanLovesCoffee) [reply src/trace/listener.ts:118]: Yeah I think that's a good idea - if `context` doesn't exist then let's skip assigning a value to `resource` altogether. Seems like we currently do that in our ruby and python layers.

Comment (hghotra) [reply src/trace/listener.ts]: Fixed.

**Pull request merged**